sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -416,6 +417,13 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
 
+  // TODO: Support Auto GRF for ESIMD once vc supports it.
+  if (isAutoGRF && !isEsimdImage) {
+    if (!CompileOpts.empty())
+      CompileOpts += " ";
+    CompileOpts += "-ze-intel-enable-auto-large-GRF-mode";
```

Comment:
Do I understand correctly, that the property is only supported by Level Zero backend? OpenCL spec has no documented support for this compile option and therefore we have no guarantee that it would be accepted by a backend.

This is one more TODO which we should note for the formal extension: clear interface for user to understand how to write portable code, i.e. how to query if a property is supported by a device? which combination of properties are invalid? etc.

Reply:
it works for opencl too for igc, but yeah it's not documented